Add long and medium term goals for the web team (#1433)
sourcegraph/about: handbook/engineering/web/index.md

# Web team

The web team owns the maintenance and expansion of our web application and code host integrations as vehicles to deliver the value of [extensions](https://docs.sourcegraph.com/extensions) to our users.

This is a large ownership area, so the team creates a focused plan each iteration, by agreeing on an appropriately small set of [iteration goals](../../../company/goals/index.md). Each goal should have more than one teammate working on it.

> The web team's current focus is documented in [the tracking issue for the current milestone](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aissue+label%3Atracking+label%3Ateam%2Fweb+is%3Aopen).

Reviewer: The goals section now feels like it is missing short term, but that is documented here. Should we move this sentence (and perhaps the previous) down to the goal section under a `### Short term` heading?

## Goals

### Long term

> ***Deliver our users the full, unique value of [extensions](https://docs.sourcegraph.com/extensions).***

Reviewer 2: Sentence nitpick: I think it should be either

```suggestion
***Deliver to our users the full, unique value of [extensions](https://docs.sourcegraph.com/extensions).***
```

or

```suggestion
***Deliver the full, unique value of [extensions](https://docs.sourcegraph.com/extensions) to our users.***
```

Reviewer 3: I prefer the second of these suggestions.

**Outcome**: Our webapp, browser extensions and native integrations are great platforms to provide our users unique value through extensions. These platforms are well-maintained, consistent, easy to setup, well-documented, well-tested, performant, and show their power through convincing extensions built on top of them. The extensions platform and API provide powerful capabilities to extension developers and a great developer experience.

> ### Medium term

Reviewer: (as discussed in web sync) In web sync you gave extra color/info on what these goals mean, and I think that would be valuable to capture here.

Author: I added extra information to each point while trying to balance it with keeping these medium-term goals high-level (as opposed to a sequence of short-term goals). Please let me know what you think!

To reach our long-term goal, we set the following medium-term goals to guide our short-term iteration plans.

> We will gradually transition our focus through these goals in the order they are defined here.

Reviewer 3: Felix and I discussed that these are not hard incremental boundaries, and work will be done in order, but there will be some overlap.

```suggestion
We will tackle these medium-term goals in order, though expect to have some work done in parallel as we progress. 
```

Author: Yeah, that's what I actually already meant here with "gradually", thanks for making that clearer!

> 1. Make the products extensions are build on (web app, code host integrations) more consistent and improve discoverability.

Reviewer: ```suggestion
1. Make the products that extensions are build on (web app, code host integrations) more consistent and improve discoverability.
```
